src/components/fields/contact-number/types.ts

```diff
@@ -0,0 +1,46 @@
+import { FormInputGroupProps } from "@lifesg/react-design-system/form/types";
+import {
+	IFrontendEngineBaseFieldJsonSchema,
+	TFrontendEngineBaseFieldJsonSchemaKeys,
+} from "../../frontend-engine/types";
+import { IYupValidationRule } from "../../frontend-engine/yup/types";
+import { InternationalCallingCodeMap } from "./data";
+
+export type TCountry = keyof typeof InternationalCallingCodeMap;
+
+interface IContactNumberProps extends FormInputGroupProps<string, unknown> {
+	country?: TCountry;
+	allowInternationalNumbers?: boolean;
+	enableSearch?: boolean;
+}
+
+export interface ISingaporeNumberValidationRule {
+	singaporeNumber?:
+		| boolean
+		| {
+				isHomeNumber: boolean;
```

Comment:
actually can explicitly state `homeNumber` to be `true` instead of `boolean` so we can ensure the devs will set either of them true